client/state/shortcodes/reducer.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { shortcodesSchema } from './schema';
+import { createReducer } from 'state/utils';
+import {
+	SHORTCODE_RECEIVE,
+	SHORTCODE_REQUEST,
+	SHORTCODE_REQUEST_FAILURE,
+	SHORTCODE_REQUEST_SUCCESS
+} from 'state/action-types';
+
+const requestingShortcode = ( state, siteId, shortcode, requesting ) => {
+	const siteShortcodes = has( state, siteId ) ? state[ siteId ] : {};
+	return {
```

Comment:
[Lodash `_.merge`](https://lodash.com/docs/4.16.4#merge) can be nice for this type of deep assign:

```js
return merge( {}, state, {
	[ siteId ]: {
		[ shortcode ]: requesting
	}
} );
```

Only time it gets you in trouble is when properties on a nested item need to be unset.

As is though, is an error thrown if you try to use the spread where the existing value would be `undefined`? Thinking:

```js
return {
	...state,
	[ siteId ]: {
		...state[ siteId ],
		[ shortcode ]: requesting
	}
};
```